components/node-labeler/cmd/run.go

```diff
@@ -74,35 +78,27 @@ var runCmd = &cobra.Command{
 			client,
 		}
 
-		c, err := controller.New("pod-watcher", mgr, controller.Options{
-			Reconciler:              r,
-			MaxConcurrentReconciles: 20,
+		filterPredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"required-by-node-labeler": "true",
+			},
 		})
+		if err != nil {
+			log.WithError(err).Fatal("unable to create predicate")
+		}
+
+		err = ctrl.NewControllerManagedBy(mgr).
+			Named("pod-watcher").
+			For(&corev1.Pod{}, builder.WithPredicates(filterPredicate)).
+			WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
```

Comment:
If we set `RequeueAfter` with `MaxConcurrentReconciles` > 1, the retry does not respect the wait time.
Now that we don't sleep in the reconcile loop, setting `MaxConcurrentReconciles` is not a problem